### DISCUSSION TITLE
Any limitations on max number of applications creation and sync?

### DISCUSSION THREAD

**Discussion by abhishek-kumar0409** [argoproj/argo-cd#17550] [General]
Hello,

What is the maximum number of application instances (create and sync) that a single instance of ArgoCD can manage?

ArgoCD will continuously poll the git repo and K8s cluster. Are there any rate limiting constraints to ensure it doesn't throttle the API server, git API server?

Isn't ArgoCD  - single point of failure ? What is advisable for production?

**Comment by farhaan-shamsee** [comment]
Does anyone has any suggestions in this regard?
